groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/compiler/ApplyGroovyTransformationProcessor.java

```diff
@@ -0,0 +1,120 @@
+            ApplyGroovyTransformation agt = e.getAnnotation(ApplyGroovyTransformation.class);
+            String[] transformations = agt.value();
+            Set<String> enableFor = new HashSet<>(Arrays.asList(agt.enable()));
+            Set<String> disableFor = new HashSet<>(Arrays.asList(agt.disable()));
+                    
+            if (transformations.length == 0) {
+                TypeMirror astType = processingEnv.getElementUtils().getTypeElement(ASTTransformation.class.getName()).asType();
+                if (!processingEnv.getTypeUtils().isAssignable(e.asType(), astType)) {
+                    throw new LayerGenerationException("Element " + e + " is not subclass of " + astType + " and transformation class is not specified by the annotation.", e); //NOI18N
+                }
+                
+                String bn = processingEnv.getElementUtils().getBinaryName((TypeElement)e).toString();
+                transformations = new String[] { bn };
+                if (enableFor.isEmpty()) {
+                    enableFor = APPLY_DEFAULT;
+                }
+            } else {
+                if (enableFor.isEmpty() && disableFor.isEmpty()) {
+                    disableFor = APPLY_DEFAULT;
+                }
+            }
+            
+            for (String mt : agt.mimeTypes()) {
+                String fnbase = "Editors/" + mt + "/Parser"; // NOI18N
+                String en = null;
+                
+                if (e instanceof TypeElement) {
+                    en = ((TypeElement)e).getQualifiedName().toString();
+                } else if (e instanceof PackageElement) {
+                    en = ((PackageElement)e).getQualifiedName().toString();
+                } else {
+                    throw new LayerGenerationException("Unexpected annotated element:" + e, e); //NOI18N
+                }
+                
+                File f = null;
+                if (!enableFor.isEmpty()) {
+                    f = generateFile(f, e, fnbase, en, enableFor, "enable", transformations); // NOI18N
+                }
+                if (!disableFor.isEmpty()) {
+                    f = generateFile(f, e, fnbase, en, disableFor, "disable", transformations); // NOI18N
+                }
+                if (f != null) {
+                    f.write();
+                }
+            }
+        }
+        return true;
+    }
+    
+    private File generateFile(File f, Element e, String fnbase, String en, Set<String> items, String att, String[] transformations) throws LayerGenerationException {
+        boolean simple = items.size() == 1 && items.contains(ApplyGroovyTransformation.APPLY_PARSE);
+        
+        if (f == null) {
+            f = layer(e).instanceFile(fnbase, en).
+                    stringvalue("instanceOf", ParsingCompilerCustomizer.class.getName()). // NOI18N
+                    methodvalue("instanceCreate", SimpleTransformationCustomizer.class.getName(), "fromLayer"); // NOI18N
```

Comment:
The fully qualified name of `SimpleTransformationCustomizer` and its `fromLayer` method becomes an API that cannot be changed. It might be better to move the factory to some class that is already part of the API...